Domain test form update

blacksponge opened the pull request:
## Purpose

Improve form accessibility

## Context

2022-11-08 team meeting 

## Changes

* move protocol and profile in the same section, at the end of the form
* use `fieldset` to delimit sections of form
* add aria-label for input that do not have a visible label
* add titles for button
* remove gradient and adjust color contrast
* remove background for "about" section
* only display profile select if several profiles are available
* use aria-describedby to link validation error to inputs
* keep at least one empty row when fetching ds from parent zone and no ds are found
* fetch data for DS and NS independently
* fix "Run test" button disabled when no data is found in parent zone
* disable the whole form instead of just the domain input and check button when waiting for a rpc response

![Screenshot 2022-11-28 at 17-02-08 Zonemaster](https://user-images.githubusercontent.com/19394895/204324319-b4ed7c42-4aa1-4078-8955-982ccfabeed5.png)

## How to test this PR

Check that the form UI visually matches the screenshot in Changes.
Check that running a test is working, the progress bar is displayed and working.

Test fix of empty response for fetch data
* For a non existing domain (e.g. `toto`) and a domain without DS (e.g. `toto.fr`)
  * Try to fetch data (NS and DS) for the domain
  * Check that a warning notification is displayed for DS (and NS for a non existing domain) but the buttons / input are not disabled
  * Check the DS (and NS for a non existing domain) forms have one empty row


Test form disabled
* Try to fetch data (NS and DS) for a existing domain
* Check that the form is disabled during the time of the query
* Check that a info notification is displayed

Test profile select only displayed if more than one choice
* Make sure to have only one profile in backend
* Load the "Run test" page
* Check that the "profile" select is not displayed
* Add a second profile in backend and restart
* Reload the "Run test" page
* Check that the "profile" select is displayed with a label, aligned with the "ipv4/ipv6" checkbox and contains the two profiles

Test aria descriptions
* Do once using empty inputs to test "required" errors and once by filling form with garbage to test "server" errors
  * Generate form errors.
  * Using [Firefox accessibility inspector](https://firefox-source-docs.mozilla.org/devtools-user/accessibility_inspector/index.html) check that each input with an error has an associated description.
    ![image](https://user-images.githubusercontent.com/19394895/204327893-7b3ad7e6-ba2f-4442-b903-89ad2b63b7fb.png)

Test aria labels
* Using the Firefox accessibility inspector again check that all buttons and inputs have a textual name.
   ![image](https://user-images.githubusercontent.com/19394895/204328806-e27c5297-1824-40ec-a00e-f078cce6a6a9.png)

Test error on fetch data
* Stop the backend
* Try to query for parent data for any domain
* Check that an error notification is displayed
* Check that the form is not disabled

pnax commented on src/app/components/form/form.component.html:
```diff
@@ -58,172 +58,182 @@ <h1 i18n>Domain name</h1>
-                {{ form.controls.profile.errors.serverError }}
-              </div>
+              <section class="col-md-5">
+                <h5 for="profile" class="form-label">Protocol</h5>
```
`for="profile"`, is this needed here? (looks like a copy/paste error)

pnax replied:
resolved since oudated

pnax commented on src/app/components/form/form.component.html:
```diff
@@ -58,172 +58,182 @@ <h1 i18n>Domain name</h1>
-              <div class="invalid-feedback" *ngIf="form.controls.profile.errors?.serverError" >
-                {{ form.controls.profile.errors.serverError }}
-              </div>
+              <section class="col-md-5">
```
suggestion: add a `margin-top` on small screen

blacksponge replied:
Done

pnax commented on src/app/components/form/form.component.html:
```diff
@@ -58,172 +58,182 @@ <h1 i18n>Domain name</h1>
+                  <button (click)="deleteRow('nameservers', i)"
+                          class="btn btn-danger delete"
+                          i18n-title
+                          title="Delete namesever">
```
typo: "namesever" -> "nameserver"

blacksponge replied:
Fixed

pnax commented on src/app/components/form/form.component.html:
```diff
@@ -58,172 +58,182 @@ <h1 i18n>Domain name</h1>
+                          (click)="addNewRow('nameservers')"
+                          class="btn btn-success add"
+                          i18n-title
+                          title="Add new namesever">
```
same typo

blacksponge replied:
Fixed

matsduf commented:
The "fetch data from parent zone" should be together with nameserver and DS.

Where will select profile be in the new setup?

If we change there, I suggest that "Delegation Signers (DS records)" is changed to "DS records" to make it less complex. I do not think there is anybody that understands "Delegation Signers" but do not know what DS records are.

blacksponge commented:
> The "fetch data from parent zone" should be together with nameserver and DS.

I will see what I can do about that, I would rather avoid having buttons at several places.

> Where will select profile be in the new setup?

In the "general" section.
![Screenshot 2022-11-23 at 18-31-20 Zonemaster](https://user-images.githubusercontent.com/19394895/203611822-b0e8b353-1dd7-420d-bf28-1f676de1697a.png)

blacksponge commented:
> If we change there, I suggest that "Delegation Signers (DS records)" is changed to "DS records" to make it less complex. I do not think there is anybody that understands "Delegation Signers" but do not know what DS records are.

Okay, I'll do that.

blacksponge commented:
> The "fetch data from parent zone" should be together with nameserver and DS.

I split the "fetch data from parent zone" in two, one for NS one for DS as it was suggested in the meeting notes. Until we split the method in backend both buttons  use same one.

(unrelated but the fetch data from parent method in backend seems to be broken in develop but not in master, I will investigate tomorrow)

blacksponge commented:
I have updated the screenshot in PR description with latest changes.

matsduf commented:
I tried to build it, but I got a lot of error messages. Could those be ignored?
```
✔ Browser application bundle generation complete.
⠋ Generating localized bundles...No translation found for "ngb.pagination.first" ("««").
No translation found for "ngb.pagination.previous" ("«").
No translation found for "ngb.pagination.next" ("»").
No translation found for "ngb.pagination.last" ("»»").
No translation found for "ngb.pagination.first-aria" ("First").
No translation found for "ngb.pagination.previous-aria" ("Previous").
No translation found for "ngb.pagination.next-aria" ("Next").
No translation found for "ngb.pagination.last-aria" ("Last").
No translation found for "ngb.progressbar.value" ("{$INTERPOLATION}").
No translation found for "1302927842198929595" ["51ec22e1dae591a194ac7448bb769dc8d49caa5f"] ("Clear domain input").
No translation found for "7937552375572060999" ["ca33ee6b36abb05c3132faae24aa707bf4393377"] ("Fetch NS from parent zone").
No translation found for "5978906015116380499" ["469bd9b4d935f4b60b0fb10af4dba592c6c14f5a"] ("DS records").
No translation found for "2736837667121738918" ["1a38bccdfebed08e1a0918aae25558b4bd561bc0"] ("Fetch DS from parent zone").
No translation found for "6439365426343089851" ["3d53f64033c4b76fdc1076ba15955d913209866c"] ("General").
No translation found for "239991585815384033" ["e6016cc16f7146d05fb4b3d8d4a9cf3aa894d8a6", "6945273904734345634"] (" Disable IPv4 ").
No translation found for "6261640190465871321" ["5fbd1b623677ff4abe8ff5a57603a17b2e5d2800", "7819258946181140854"] (" Disable IPv6 ").
No translation found for "6534302992540929721" ["c4d8a11a168d1bed37bc5a1cf302dfeb2bc62d6c"] ("Nameserver").
No translation found for "2483018961011868950" ["e8096ea93b6627c620ffbb208d7ad49235ffc2fd"] ("IP address").
No translation found for "2802122512502430271" ["9e654b8e014437daf59581aeab2425a7cd0c0007"] ("Delete nameserver").
No translation found for "5409217774537333885" ["8e28ba95684e8704a43f67ac1d33bb2e2a9c54fd"] ("Add new nameserver").
No translation found for "9012648368816784794" ["94f7a0a482e16dafb093eff9c64bdffc40dc0636", "3850671844440976756"] (" Nameserver required ").
No translation found for "4004660444874033759" ["a35d390f659821874ed1913ccf54b9a2d0d79b2a"] ("Delete DS record").
No translation found for "2877617269500311707" ["ea59270a53cba963d581a60e32c464a0339e2c12"] ("Add new DS record").
No translation found for "2963939280706300945" ["ea9a4cfd79cc6b5e7be4c0f52701d54b127d89f6", "6512926724073182723"] (" Keytag required ").
No translation found for "8390821188168588727" ["dfb71f4aa0784b7ddcc0511e4d2ed664bbf24830", "8013631368152152203"] (" Algorithm required ").
No translation found for "4535340806827339238" ["a8ccbfc31e1cfe3c97e9d017ca41a8f091d0a8d3", "618067252229511529"] (" Digest type required ").
No translation found for "2636921205173022218" ["aafb5b89839f9af24b0f7b61208d25361f11b293", "6799425788893555852"] (" Digest required ").
No translation found for "3998534068662055928" ["41d60762c584ced08c7f64af8b34c35ce9906f3e", "1833216682015604379"] (" Choose at least one protocol ").
No translation found for "ngb.pagination.first" ("««").
No translation found for "ngb.pagination.previous" ("«").
No translation found for "ngb.pagination.next" ("»").
No translation found for "ngb.pagination.last" ("»»").
No translation found for "ngb.pagination.first-aria" ("First").
No translation found for "ngb.pagination.previous-aria" ("Previous").
No translation found for "ngb.pagination.next-aria" ("Next").
No translation found for "ngb.pagination.last-aria" ("Last").
No translation found for "ngb.progressbar.value" ("{$INTERPOLATION}").
No translation found for "1302927842198929595" ["51ec22e1dae591a194ac7448bb769dc8d49caa5f"] ("Clear domain input").
No translation found for "7937552375572060999" ["ca33ee6b36abb05c3132faae24aa707bf4393377"] ("Fetch NS from parent zone").
No translation found for "5978906015116380499" ["469bd9b4d935f4b60b0fb10af4dba592c6c14f5a"] ("DS records").
No translation found for "2736837667121738918" ["1a38bccdfebed08e1a0918aae25558b4bd561bc0"] ("Fetch DS from parent zone").
No translation found for "6439365426343089851" ["3d53f64033c4b76fdc1076ba15955d913209866c"] ("General").
No translation found for "239991585815384033" ["e6016cc16f7146d05fb4b3d8d4a9cf3aa894d8a6", "6945273904734345634"] (" Disable IPv4 ").
No translation found for "6261640190465871321" ["5fbd1b623677ff4abe8ff5a57603a17b2e5d2800", "7819258946181140854"] (" Disable IPv6 ").
No translation found for "6534302992540929721" ["c4d8a11a168d1bed37bc5a1cf302dfeb2bc62d6c"] ("Nameserver").
No translation found for "2483018961011868950" ["e8096ea93b6627c620ffbb208d7ad49235ffc2fd"] ("IP address").
No translation found for "2802122512502430271" ["9e654b8e014437daf59581aeab2425a7cd0c0007"] ("Delete nameserver").
No translation found for "5409217774537333885" ["8e28ba95684e8704a43f67ac1d33bb2e2a9c54fd"] ("Add new nameserver").
No translation found for "9012648368816784794" ["94f7a0a482e16dafb093eff9c64bdffc40dc0636", "3850671844440976756"] (" Nameserver required ").
No translation found for "4004660444874033759" ["a35d390f659821874ed1913ccf54b9a2d0d79b2a"] ("Delete DS record").
No translation found for "2877617269500311707" ["ea59270a53cba963d581a60e32c464a0339e2c12"] ("Add new DS record").
No translation found for "2963939280706300945" ["ea9a4cfd79cc6b5e7be4c0f52701d54b127d89f6", "6512926724073182723"] (" Keytag required ").
No translation found for "8390821188168588727" ["dfb71f4aa0784b7ddcc0511e4d2ed664bbf24830", "8013631368152152203"] (" Algorithm required ").
No translation found for "4535340806827339238" ["a8ccbfc31e1cfe3c97e9d017ca41a8f091d0a8d3", "618067252229511529"] (" Digest type required ").
No translation found for "2636921205173022218" ["aafb5b89839f9af24b0f7b61208d25361f11b293", "6799425788893555852"] (" Digest required ").
No translation found for "3998534068662055928" ["41d60762c584ced08c7f64af8b34c35ce9906f3e", "1833216682015604379"] (" Choose at least one protocol ").
No translation found for "ngb.pagination.first" ("««").
No translation found for "ngb.pagination.previous" ("«").
No translation found for "ngb.pagination.next" ("»").
No translation found for "ngb.pagination.last" ("»»").
No translation found for "ngb.pagination.first-aria" ("First").
No translation found for "ngb.pagination.previous-aria" ("Previous").
No translation found for "ngb.pagination.next-aria" ("Next").
No translation found for "ngb.pagination.last-aria" ("Last").
No translation found for "ngb.progressbar.value" ("{$INTERPOLATION}").
No translation found for "1302927842198929595" ["51ec22e1dae591a194ac7448bb769dc8d49caa5f"] ("Clear domain input").
No translation found for "7937552375572060999" ["ca33ee6b36abb05c3132faae24aa707bf4393377"] ("Fetch NS from parent zone").
No translation found for "5978906015116380499" ["469bd9b4d935f4b60b0fb10af4dba592c6c14f5a"] ("DS records").
No translation found for "2736837667121738918" ["1a38bccdfebed08e1a0918aae25558b4bd561bc0"] ("Fetch DS from parent zone").
No translation found for "6439365426343089851" ["3d53f64033c4b76fdc1076ba15955d913209866c"] ("General").
No translation found for "239991585815384033" ["e6016cc16f7146d05fb4b3d8d4a9cf3aa894d8a6", "6945273904734345634"] (" Disable IPv4 ").
No translation found for "6261640190465871321" ["5fbd1b623677ff4abe8ff5a57603a17b2e5d2800", "7819258946181140854"] (" Disable IPv6 ").
No translation found for "6534302992540929721" ["c4d8a11a168d1bed37bc5a1cf302dfeb2bc62d6c"] ("Nameserver").
No translation found for "2483018961011868950" ["e8096ea93b6627c620ffbb208d7ad49235ffc2fd"] ("IP address").
No translation found for "2802122512502430271" ["9e654b8e014437daf59581aeab2425a7cd0c0007"] ("Delete nameserver").
No translation found for "5409217774537333885" ["8e28ba95684e8704a43f67ac1d33bb2e2a9c54fd"] ("Add new nameserver").
No translation found for "9012648368816784794" ["94f7a0a482e16dafb093eff9c64bdffc40dc0636", "3850671844440976756"] (" Nameserver required ").
No translation found for "4004660444874033759" ["a35d390f659821874ed1913ccf54b9a2d0d79b2a"] ("Delete DS record").
No translation found for "2877617269500311707" ["ea59270a53cba963d581a60e32c464a0339e2c12"] ("Add new DS record").
No translation found for "2963939280706300945" ["ea9a4cfd79cc6b5e7be4c0f52701d54b127d89f6", "6512926724073182723"] (" Keytag required ").
No translation found for "8390821188168588727" ["dfb71f4aa0784b7ddcc0511e4d2ed664bbf24830", "8013631368152152203"] (" Algorithm required ").
No translation found for "4535340806827339238" ["a8ccbfc31e1cfe3c97e9d017ca41a8f091d0a8d3", "618067252229511529"] (" Digest type required ").
No translation found for "2636921205173022218" ["aafb5b89839f9af24b0f7b61208d25361f11b293", "6799425788893555852"] (" Digest required ").
No translation found for "3998534068662055928" ["41d60762c584ced08c7f64af8b34c35ce9906f3e", "1833216682015604379"] (" Choose at least one protocol ").
No translation found for "ngb.pagination.first" ("««").
No translation found for "ngb.pagination.previous" ("«").
No translation found for "ngb.pagination.next" ("»").
No translation found for "ngb.pagination.last" ("»»").
No translation found for "ngb.pagination.first-aria" ("First").
No translation found for "ngb.pagination.previous-aria" ("Previous").
No translation found for "ngb.pagination.next-aria" ("Next").
No translation found for "ngb.pagination.last-aria" ("Last").
No translation found for "ngb.progressbar.value" ("{$INTERPOLATION}").
No translation found for "1302927842198929595" ["51ec22e1dae591a194ac7448bb769dc8d49caa5f"] ("Clear domain input").
No translation found for "7937552375572060999" ["ca33ee6b36abb05c3132faae24aa707bf4393377"] ("Fetch NS from parent zone").
No translation found for "5978906015116380499" ["469bd9b4d935f4b60b0fb10af4dba592c6c14f5a"] ("DS records").
No translation found for "2736837667121738918" ["1a38bccdfebed08e1a0918aae25558b4bd561bc0"] ("Fetch DS from parent zone").
No translation found for "6439365426343089851" ["3d53f64033c4b76fdc1076ba15955d913209866c"] ("General").
No translation found for "239991585815384033" ["e6016cc16f7146d05fb4b3d8d4a9cf3aa894d8a6", "6945273904734345634"] (" Disable IPv4 ").
No translation found for "6261640190465871321" ["5fbd1b623677ff4abe8ff5a57603a17b2e5d2800", "7819258946181140854"] (" Disable IPv6 ").
No translation found for "6534302992540929721" ["c4d8a11a168d1bed37bc5a1cf302dfeb2bc62d6c"] ("Nameserver").
No translation found for "2483018961011868950" ["e8096ea93b6627c620ffbb208d7ad49235ffc2fd"] ("IP address").
No translation found for "2802122512502430271" ["9e654b8e014437daf59581aeab2425a7cd0c0007"] ("Delete nameserver").
No translation found for "5409217774537333885" ["8e28ba95684e8704a43f67ac1d33bb2e2a9c54fd"] ("Add new nameserver").
No translation found for "9012648368816784794" ["94f7a0a482e16dafb093eff9c64bdffc40dc0636", "3850671844440976756"] (" Nameserver required ").
No translation found for "4004660444874033759" ["a35d390f659821874ed1913ccf54b9a2d0d79b2a"] ("Delete DS record").
No translation found for "2877617269500311707" ["ea59270a53cba963d581a60e32c464a0339e2c12"] ("Add new DS record").
No translation found for "2963939280706300945" ["ea9a4cfd79cc6b5e7be4c0f52701d54b127d89f6", "6512926724073182723"] (" Keytag required ").
No translation found for "8390821188168588727" ["dfb71f4aa0784b7ddcc0511e4d2ed664bbf24830", "8013631368152152203"] (" Algorithm required ").
No translation found for "4535340806827339238" ["a8ccbfc31e1cfe3c97e9d017ca41a8f091d0a8d3", "618067252229511529"] (" Digest type required ").
No translation found for "2636921205173022218" ["aafb5b89839f9af24b0f7b61208d25361f11b293", "6799425788893555852"] (" Digest required ").
No translation found for "3998534068662055928" ["41d60762c584ced08c7f64af8b34c35ce9906f3e", "1833216682015604379"] (" Choose at least one protocol ").
No translation found for "ngb.pagination.first" ("««").
No translation found for "ngb.pagination.previous" ("«").
No translation found for "ngb.pagination.next" ("»").
No translation found for "ngb.pagination.last" ("»»").
No translation found for "ngb.pagination.first-aria" ("First").
No translation found for "ngb.pagination.previous-aria" ("Previous").
No translation found for "ngb.pagination.next-aria" ("Next").
No translation found for "ngb.pagination.last-aria" ("Last").
No translation found for "ngb.progressbar.value" ("{$INTERPOLATION}").
No translation found for "1302927842198929595" ["51ec22e1dae591a194ac7448bb769dc8d49caa5f"] ("Clear domain input").
No translation found for "7937552375572060999" ["ca33ee6b36abb05c3132faae24aa707bf4393377"] ("Fetch NS from parent zone").
No translation found for "5978906015116380499" ["469bd9b4d935f4b60b0fb10af4dba592c6c14f5a"] ("DS records").
No translation found for "2736837667121738918" ["1a38bccdfebed08e1a0918aae25558b4bd561bc0"] ("Fetch DS from parent zone").
No translation found for "6439365426343089851" ["3d53f64033c4b76fdc1076ba15955d913209866c"] ("General").
No translation found for "239991585815384033" ["e6016cc16f7146d05fb4b3d8d4a9cf3aa894d8a6", "6945273904734345634"] (" Disable IPv4 ").
No translation found for "6261640190465871321" ["5fbd1b623677ff4abe8ff5a57603a17b2e5d2800", "7819258946181140854"] (" Disable IPv6 ").
No translation found for "6534302992540929721" ["c4d8a11a168d1bed37bc5a1cf302dfeb2bc62d6c"] ("Nameserver").
No translation found for "2483018961011868950" ["e8096ea93b6627c620ffbb208d7ad49235ffc2fd"] ("IP address").
No translation found for "2802122512502430271" ["9e654b8e014437daf59581aeab2425a7cd0c0007"] ("Delete nameserver").
No translation found for "5409217774537333885" ["8e28ba95684e8704a43f67ac1d33bb2e2a9c54fd"] ("Add new nameserver").
No translation found for "9012648368816784794" ["94f7a0a482e16dafb093eff9c64bdffc40dc0636", "3850671844440976756"] (" Nameserver required ").
No translation found for "4004660444874033759" ["a35d390f659821874ed1913ccf54b9a2d0d79b2a"] ("Delete DS record").
No translation found for "2877617269500311707" ["ea59270a53cba963d581a60e32c464a0339e2c12"] ("Add new DS record").
No translation found for "2963939280706300945" ["ea9a4cfd79cc6b5e7be4c0f52701d54b127d89f6", "6512926724073182723"] (" Keytag required ").
No translation found for "8390821188168588727" ["dfb71f4aa0784b7ddcc0511e4d2ed664bbf24830", "8013631368152152203"] (" Algorithm required ").
No translation found for "4535340806827339238" ["a8ccbfc31e1cfe3c97e9d017ca41a8f091d0a8d3", "618067252229511529"] (" Digest type required ").
No translation found for "2636921205173022218" ["aafb5b89839f9af24b0f7b61208d25361f11b293", "6799425788893555852"] (" Digest required ").
No translation found for "3998534068662055928" ["41d60762c584ced08c7f64af8b34c35ce9906f3e", "1833216682015604379"] (" Choose at least one protocol ").
No translation found for "ngb.pagination.first" ("««").
No translation found for "ngb.pagination.previous" ("«").
No translation found for "ngb.pagination.next" ("»").
No translation found for "ngb.pagination.last" ("»»").
No translation found for "ngb.pagination.first-aria" ("First").
No translation found for "ngb.pagination.previous-aria" ("Previous").
No translation found for "ngb.pagination.next-aria" ("Next").
No translation found for "ngb.pagination.last-aria" ("Last").
No translation found for "ngb.progressbar.value" ("{$INTERPOLATION}").
No translation found for "1302927842198929595" ["51ec22e1dae591a194ac7448bb769dc8d49caa5f"] ("Clear domain input").
No translation found for "7937552375572060999" ["ca33ee6b36abb05c3132faae24aa707bf4393377"] ("Fetch NS from parent zone").
No translation found for "5978906015116380499" ["469bd9b4d935f4b60b0fb10af4dba592c6c14f5a"] ("DS records").
No translation found for "2736837667121738918" ["1a38bccdfebed08e1a0918aae25558b4bd561bc0"] ("Fetch DS from parent zone").
No translation found for "6439365426343089851" ["3d53f64033c4b76fdc1076ba15955d913209866c"] ("General").
No translation found for "239991585815384033" ["e6016cc16f7146d05fb4b3d8d4a9cf3aa894d8a6", "6945273904734345634"] (" Disable IPv4 ").
No translation found for "6261640190465871321" ["5fbd1b623677ff4abe8ff5a57603a17b2e5d2800", "7819258946181140854"] (" Disable IPv6 ").
No translation found for "6534302992540929721" ["c4d8a11a168d1bed37bc5a1cf302dfeb2bc62d6c"] ("Nameserver").
No translation found for "2483018961011868950" ["e8096ea93b6627c620ffbb208d7ad49235ffc2fd"] ("IP address").
No translation found for "2802122512502430271" ["9e654b8e014437daf59581aeab2425a7cd0c0007"] ("Delete nameserver").
No translation found for "5409217774537333885" ["8e28ba95684e8704a43f67ac1d33bb2e2a9c54fd"] ("Add new nameserver").
No translation found for "9012648368816784794" ["94f7a0a482e16dafb093eff9c64bdffc40dc0636", "3850671844440976756"] (" Nameserver required ").
No translation found for "4004660444874033759" ["a35d390f659821874ed1913ccf54b9a2d0d79b2a"] ("Delete DS record").
No translation found for "2877617269500311707" ["ea59270a53cba963d581a60e32c464a0339e2c12"] ("Add new DS record").
No translation found for "2963939280706300945" ["ea9a4cfd79cc6b5e7be4c0f52701d54b127d89f6", "6512926724073182723"] (" Keytag required ").
No translation found for "8390821188168588727" ["dfb71f4aa0784b7ddcc0511e4d2ed664bbf24830", "8013631368152152203"] (" Algorithm required ").
No translation found for "4535340806827339238" ["a8ccbfc31e1cfe3c97e9d017ca41a8f091d0a8d3", "618067252229511529"] (" Digest type required ").
No translation found for "2636921205173022218" ["aafb5b89839f9af24b0f7b61208d25361f11b293", "6799425788893555852"] (" Digest required ").
No translation found for "3998534068662055928" ["41d60762c584ced08c7f64af8b34c35ce9906f3e", "1833216682015604379"] (" Choose at least one protocol ").
✔ Localized bundle generation complete.
✔ Copying assets complete.
⠋ Generating index html...1 rules skipped due to selector errors:
  legend+* -> Cannot read properties of undefined (reading 'type')
⠋ Generating index html...1 rules skipped due to selector errors:
  legend+* -> Cannot read properties of undefined (reading 'type')
⠋ Generating index html...1 rules skipped due to selector errors:
  legend+* -> Cannot read properties of undefined (reading 'type')
⠋ Generating index html...1 rules skipped due to selector errors:
  legend+* -> Cannot read properties of undefined (reading 'type')
⠋ Generating index html...1 rules skipped due to selector errors:
  legend+* -> Cannot read properties of undefined (reading 'type')
⠋ Generating index html...1 rules skipped due to selector errors:
  legend+* -> Cannot read properties of undefined (reading 'type')
⠋ Generating index html...1 rules skipped due to selector errors:
  legend+* -> Cannot read properties of undefined (reading 'type')
✔ Index html generation complete.

Initial Chunk Files           | Names         |  Raw Size | Estimated Transfer Size
main.d8c2ed1dc5b17f77.js      | main          | 649.67 kB |               142.91 kB
styles.4f3a77c5ecd4e2c3.css   | styles        | 229.10 kB |                26.74 kB
polyfills.8c668a65c41cb99e.js | polyfills     |  33.06 kB |                10.67 kB
runtime.faf8f5694cfc0105.js   | runtime       |   1.06 kB |               598 bytes

                              | Initial Total | 912.88 kB |               180.90 kB

Build at: 2022-11-23T22:32:46.586Z - Hash: 65887bb4a356d992 - Time: 53599ms

Warning: /home/ubuntu/git/zonemaster-gui/src/app/components/navigation/navigation.component.ts depends on 'rxjs/internal/operators'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies
```

blacksponge commented:
> I tried to build it, but I got a lot of error messages. Could those be ignored?

You can ignore those for the moment. They are from strings not yet translated.

matsduf commented:
I can test iis.se, but when I "open" options and press "fetch NS" I get an error:
![image](https://user-images.githubusercontent.com/5430733/203743673-475b692b-dc15-45ed-a77d-b22e4e3cf1f7.png)

matsduf commented on src/app/components/form/form.component.html:
```diff
@@ -53,182 +59,232 @@ <h1 i18n>Domain name</h1>
-                <div i18n class="invalid-feedback" *ngIf="ds_info.controls.keytag.errors?.required" >
-                  Keytag required
+                <div id="protocol-error" i18n class="invalid-feedback" *ngIf="form.errors?.noProtocol">
+                  Choose at least one protocol
```
Suggest: "Choose at least one protocol" --> "Cannot disable both IPv4 and IPv6"

blacksponge replied:
Done

blacksponge commented:
This come from a regression in engine / backend see my comment above:

> (unrelated but the fetch data from parent method in backend seems to be broken in develop but not in master, I will investigate tomorrow)

matsduf reviewed (changes requested):
Looks OK, but fetching NS or DS do not work (see other comment).

matsduf commented on src/app/components/form/form.component.html:
```diff
@@ -53,182 +59,232 @@ <h1 i18n>Domain name</h1>
+            <legend i18n>General</legend>
+            <div class="row">
+              <fieldset aria-describedby="protocol-error" class="col-md-5 form-col">
+                <legend for="profile">Protocol</legend>
```
"Protocol" does not add anything. It is better to remove.

blacksponge replied:
I am not sure about that. It adds a grouping label for the two check box and add visual balance when the profile select is present.

If I remove it then I if don't do any think the checkbox are not aligned with the select any more:
![Screenshot 2022-11-24 at 11-27-23 Zonemaster](https://user-images.githubusercontent.com/19394895/203761112-e9eb1f4d-e37b-4025-8f6c-4fb97cad25a7.png)

If I add some margin to the checkbox it can work when the select is present
![Screenshot 2022-11-24 at 11-27-11 Zonemaster](https://user-images.githubusercontent.com/19394895/203761215-07b61530-8e7b-4268-8ee6-c5112ab8b106.png)

but introduced a big gap when the select is hidden

![Screenshot 2022-11-24 at 11-26-51 Zonemaster](https://user-images.githubusercontent.com/19394895/203761267-562a2a4e-e195-4d6e-858b-6d9e99f04b8a.png)

tgreenx replied:
I think this is useful to have as well.

matsduf replied:
> I think this is useful to have as well.

For whom do you think it adds anything?

matsduf replied:
> I am not sure about that. It adds a grouping label for the two check boxes and add visual balance when the profile select is present.

Would it work with 
```
Profile: [default                      ^]
```
aligned with the check boxes?

tgreenx replied:
If the alignement issue can be resolved, this is fine for me to remove the title "Protocol"

blacksponge replied:
I will test that.

matsduf reviewed (dismissed):
My comment on "Protocol" remains, but it is not blocking.

tgreenx reviewed (commented):
If fetching data from parent (NS or DS) fails, the "Check" button greys out and can't be used again, preventing start of tests, even when changing options or domain name.

matsduf commented:
> If fetching data from parent (NS or DS) fails, the "Check" button greys out and can't be used again, preventing start of tests, even when changing options or domain name.

I also noticed that. Will that go away completely when the issue is resolved in Engine, or will it remain if the domain name is unreachable?

blacksponge commented:
> I also noticed that. Will that go away completely when the issue is resolved in Engine, or will it remain if the domain name is unreachable?

It seems that the button will remain disable, this issue is older than the issue in Engine (it is the same on zonemaster.net). I will add a fix for it in this PR.

blacksponge commented:
> I also noticed that. Will that go away completely when the issue is resolved in Engine, or will it remain if the domain name is unreachable?

It should be fixed now.

matsduf commented:
@blacksponge, I suggest that you rebase this on current develop.

blacksponge commented:
While writing the how to test section I noticed a few implementation mistake, now fixed.

tgreenx reviewed (commented):
When I use the button "Fetch X from parent zone" (NS or DS) and nothing is found, the error message at the top of the screen "No data found for the zone" is not visible (it is somehow hidden at the very top,  out of screen).

blacksponge commented:
> When I use the button "Fetch X from parent zone" (NS or DS) and nothing is found, the error message at the top of the screen "No data found for the zone" is not visible (it is somehow hidden at the very top, out of screen).

I created https://github.com/zonemaster/zonemaster-gui/pull/369 to fix the issue.

mattias-p commented:
I've successfully release tested this for v2022.2. I found a couple of minor things we may want to tweak. I'll create issues for those and link them here.
